src/amo/components/CollectionManager/index.js

```diff
@@ -88,16 +99,30 @@ export class CollectionManagerBase extends React.Component<Props, State> {
       return;
     }
 
-    dispatch(updateCollection({
-      collectionSlug: collection.slug,
-      defaultLocale: collection.defaultLocale,
+    if (!this.state.slug) {
+      errorHandler.addMessage(
+        i18n.gettext('Collection slug cannot be blank')
+      );
+      return;
+    }
+
+    const payload: Object = {
```

Comment:
I tried to make this a type more specific than `Object` but Flow had me running in circles. Again it's a problem with all the optional/mandatory types needed for the two different actions. I tried to solve it by adding calls to `invariant` into the code here for the non-optional parameters, but this time that didn't work. Flow still got mad about some parameters possibly being null, even though I guaranteed they wouldn't be with `invariant`.  Perhaps that was because I have an object here, `payload`, so my calls to `invariant` looked like `invariant(payload.slug)` as opposed to just `invariant(slug)`, and maybe Flow couldn't figure out what I was doing.